criu/pagemap-cache.c

```diff
@@ -50,10 +51,19 @@ int pmc_init(pmc_t *pmc, pid_t pid, const struct list_head *vma_head, size_t siz
 	pmc->pid = pid;
 	pmc->map_len = PAGEMAP_LEN(map_size);
 	pmc->vma_head = vma_head;
+	pmc->regs_max_len = (map_size / PAGE_SIZE + 1) / 2;
```

Comment:
Won't this be huge for a big mapping (like ASan or TSan make)? If the cache refill call would be pushed to should_dump_page(), the pagemap cache could be made constant in size as the ioctl() can continue also from an address inside a VMA.

(I guess that since PAGEMAP_SCAN will also get to return SOFT_DIRTY bit, the special handling won't be needed.)